src/dataframe/dataframe.jl

```diff
@@ -190,21 +190,20 @@ struct DataFrame <: AbstractDataFrame
 
         # we write into columns as we know that it is guaranteed
         # that it was freshly allocated in the outer constructor
-        for (i, col) in enumerate(columns)
-            # check for vectors first as they are most common
-            if col isa AbstractRange
-                columns[i] = collect(col)
-            elseif col isa AbstractVector
-                columns[i] = copycols ? copy(col) : col
-            elseif col isa Union{AbstractArray{<:Any, 0}, Ref}
-                x = col[]
-                columns[i] = fill!(Tables.allocatecolumn(typeof(x), len), x)
+        @static if VERSION >= v"1.4"
+            if Threads.nthreads() > 1 && len >= 1_000_000
+                @sync for i in eachindex(columns)
+                    @async Threads.@spawn columns[i] = _preprocess_column(columns[i],
```

Comment:
AFAIK `@async` isn't needed, it creates another non-parellelizable task wrapping the one created by `@spawn`.

Reply:
@nalimilan is correct. Just `Threads.@spawn` is needed.

Reply:
I have added it because it was in the example @quinnj gave :smile:. I have removed it.

Reply:
Whoops!